fms/utils/generation.py

```diff
@@ -130,7 +130,8 @@ def truncate_after_eos(result, eos_token_id):
     """
     if eos_token_id is None:
         return result
-
+    if isinstance(eos_token_id, list) and len(eos_token_id) == 1:
+        eos_token_id = eos_token_id[0]
```

Comment:
Let's instead fix the call-site in the inference.py script:

https://github.com/foundation-model-stack/foundation-model-stack/blob/main/scripts/inference.py#L193

while we're at it, that call-site can be updated to use `tokenizer.eos_token_id` instead of the hard-coded `<\s>`.

we should also update the type annotation on this function to help avoid such bugs.

Reply:
`tokenizer.eos_token_id` is an integer, so I am not able to directly replace it for '<\s>'. is `str(tokenizer.eos_token_id)` the same thing as the string? it compiles but I'm not sure if they are the same.

Reply:
you have the integer and the tokenizer, so you can covert back and forth. I think in this particular case we want the integer, so we'd be removing the tokenizer call